frontend/src/components/ColorLegend.svelte

```diff
@@ -1,28 +1,96 @@
 <script lang="ts">
-    export let colors;
+    export let styles;
+    import G6, { Graph } from "@antv/g6";
+    import { onMount } from "svelte";
+    import { createEventDispatcher } from "svelte";
+
+    const options = {
+        container: "mountNode",
+        width: 280,
+        height: 220,
+        workerEnabled: true,
+        nodeSize: 20,
+    };
+    const dispatch = createEventDispatcher();
+
+    let data = {};
+    let array = [];
+    styles.map((event) => {
+        let node = {
+            id: event[0],
+            size: 10,
+            type: event[1].Shape,
+            style: {
+                fill: event[1].Color,
+                stroke: event[1].Color,
+            },
+            label: event[1].Acronym,
+            labelCfg: {
+                style: {
+                    fill: event[1].Color,
+                },
+                position: "bottom",
+                offset: 10,
+            },
+        };
+        array.push(node);
+    });
+    data = { nodes: array };
+
+    let container: HTMLElement;
+    let graph: Graph | null;
+    export const reset = () => {
+        graph?.changeData({});
+        graph?.render();
+        dispatch("nodeselected", null);
+    };
+
+    export const resizeGraph = () => {
```

Comment:
there seems to be a bit of duplicated code here, could we make the g6 graph into more of a generic component?